sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -995,7 +1019,7 @@ abstract class GpuBaseAggregateMeta[INPUT <: SparkPlan](
   override def convertToGpu(): GpuExec = {
     GpuHashAggregateExec(
       requiredChildDistributionExpressions.map(_.map(_.convertToGpu())),
-      groupingExpressions.map(_.convertToGpu()),
+      groupingExpressions.map(_.convertToGpu()).asInstanceOf[Seq[NamedExpression]],
```

Comment:
Because of type erasure (how java does generics) the sequence could contain things that are not NamedExpressions and the cast would happily pass. I would prefer to have us do something more like
```
groupingExpressions.map(_.convertToGpu().asInstanceOf[NamedExpression])
```